corehq/apps/users/user_data.py

```diff
@@ -31,27 +31,44 @@ def lazy_init(cls, couch_user, domain):
         raw_user_data = couch_user.to_json().get('user_data', {}).copy()
         raw_user_data.pop(COMMCARE_PROJECT, None)
         profile_id = raw_user_data.pop(PROFILE_SLUG, None)
-        sql_data, _ = SQLUserData.objects.get_or_create(
-            user_id=couch_user.user_id,
-            domain=domain,
-            defaults={
-                'data': raw_user_data,
-                'django_user': couch_user.get_django_user,
-                'profile_id': profile_id,
-            }
-        )
+
+        try:
+            sql_data = SQLUserData.objects.get(user_id=couch_user.user_id, domain=domain)
+        except SQLUserData.DoesNotExist:
+            if not (raw_user_data or profile_id):
+                # Don't bother saving anything to the DB
+                return cls({}, couch_user, domain)
+
+            sql_data = SQLUserData.objects.create(
+                user_id=couch_user.user_id,
+                domain=domain,
+                data=raw_user_data,
+                django_user=couch_user.get_django_user(),
+                profile_id=profile_id,
+            )
+
         return cls(sql_data.data, couch_user, domain, profile_id=sql_data.profile_id)
 
+    @transaction.atomic
     def save(self):
-        SQLUserData.objects.update_or_create(
-            user_id=self._couch_user.user_id,
-            domain=self.domain,
-            defaults={
-                'data': self._local_to_user,
-                'django_user': self._couch_user.get_django_user,
-                'profile_id': self.profile_id,
-            },
-        )
+        try:
+            sql_data = (SQLUserData.objects
```

Comment:
what are the () on the outside for?

Reply:
So it can span multiple lines - you'd get a syntax error without them.